src/main/java/org/skife/jdbi/v2/BeanMapper.java

```diff
@@ -81,55 +78,18 @@ public T map(int row, ResultSet rs, StatementContext ctx)
                 Class type = descriptor.getPropertyType();
 
                 Object value;
-
-                if (type.isAssignableFrom(Boolean.class) || type.isAssignableFrom(boolean.class)) {
-                    value = rs.getBoolean(i);
-                }
-                else if (type.isAssignableFrom(Byte.class) || type.isAssignableFrom(byte.class)) {
-                    value = rs.getByte(i);
-                }
-                else if (type.isAssignableFrom(Short.class) || type.isAssignableFrom(short.class)) {
-                    value = rs.getShort(i);
-                }
-                else if (type.isAssignableFrom(Integer.class) || type.isAssignableFrom(int.class)) {
-                    value = rs.getInt(i);
-                }
-                else if (type.isAssignableFrom(Long.class) || type.isAssignableFrom(long.class)) {
-                    value = rs.getLong(i);
-                }
-                else if (type.isAssignableFrom(Float.class) || type.isAssignableFrom(float.class)) {
-                    value = rs.getFloat(i);
-                }
-                else if (type.isAssignableFrom(Double.class) || type.isAssignableFrom(double.class)) {
-                    value = rs.getDouble(i);
-                }
-                else if (type.isAssignableFrom(BigDecimal.class)) {
-                    value = rs.getBigDecimal(i);
-                }
-                else if (type.isAssignableFrom(Timestamp.class)) {
-                    value = rs.getTimestamp(i);
-                }
-                else if (type.isAssignableFrom(Time.class)) {
-                    value = rs.getTime(i);
-                }
-                else if (type.isAssignableFrom(Date.class)) {
-                    value = rs.getDate(i);
-                }
-                else if (type.isAssignableFrom(String.class)) {
-                    value = rs.getString(i);
+                ResultColumnMapper mapper = ctx.columnMapperFor(type);
+                if (mapper != null) {
+                    value = mapper.mapColumn(rs, i, ctx);
                 }
                 else if (type.isEnum()) {
                     String str = rs.getString(i);
```

Comment:
@stevenschlansker So I found something that may impact design on enums: `BuiltInArgumentFactory` implicitly encodes enum arguments by their name. Because `Foreman` eagerly registers the built-ins on construction, clients cannot override this with their own alternative arguments.

We have to choose between:
* Sticking with `name()`-only encoding. In that case we should drop `EnumColumnMapper.byOrdinal(Class)` and `EnumColumnMapperFactory.BY_ORDINAL` since they are moot.
* Make enum encoding configurable. This would require making `Foreman.waffle` use `BuiltInArgumentFactory` as a last-ditch fallback rather than eagerly adding it to the factory list on construction--similar to what the mapping registry does with built-in mappers.

I prefer the latter.

In either case, it now makes sense for `EnumColumnMapperFactory.BY_NAME` to be included in the built-in column mappers, for parity with the argument factory registry. If we pick the latter option above, then clients would be expected both an argument factory and a column mapper factory if they wish to override enum encoding.